`org.jboss.jandex:jandex-maven-plugin:1.2.3:jandex` fails on Java 21

# Description

**Registry
Version**: `2.6.2.Final`

The `org.jboss.jandex:jandex-maven-plugin` `v1.2.3` module defined in the latest `2.*` version (`2.6.2.Final`) fails to build on Java 21.

This is the same problem [as discovered in the `apache/hop` project](https://github.com/apache/hop/issues/3637), solved by updating the dependency to `io.smallrye:jandex-maven-plugin` `v3.1.8` ([`3.2.0+` changes the index version, causing problems with quarkus](https://github.com/quarkusio/quarkus/issues/40739#issuecomment-2122639055)).

This update has already been made on `main` (#4656), hence this would merely be a backport of that update to the `2.*` versions.
I see #5079 did not mention this plugin. I am also noticing this robot auto-update is only done on `2.5.x`, not `2.6.x`

# Logs
```
#0 383.3 [ERROR] Failed to execute goal org.jboss.jandex:jandex-maven-plugin:1.2.3:jandex (make-index) on project apicurio-registry-utils-tools: Execution make-index of goal org.jboss.jandex:jandex-maven-plugin:1.2.3:jandex failed: Cannot read the array length because "bytes" is null -> [Help 1]
#0 383.3 [ERROR] Failed to execute goal org.jboss.jandex:jandex-maven-plugin:1.2.3:jandex (make-index) on project apicurio-registry-utils-kafka: Execution make-index of goal org.jboss.jandex:jandex-maven-plugin:1.2.3:jandex failed: Cannot read the array length because "bytes" is null -> [Help 1]
#0 383.3 [ERROR] Failed to execute goal org.jboss.jandex:jandex-maven-plugin:1.2.3:jandex (make-index) on project apicurio-registry-common: Execution make-index of goal org.jboss.jandex:jandex-maven-plugin:1.2.3:jandex failed: Cannot read the array length because "bytes" is null -> [Help 1]
```

--- a/app/pom.xml
+++ b/app/pom.xml
@@ -567,7 +567,7 @@
             <build>
                 <plugins>
                     <plugin>
-                        <groupId>org.jboss.jandex</groupId>
+                        <groupId>io.smallrye</groupId>
                         <artifactId>jandex-maven-plugin</artifactId>
                         <executions>
                             <execution>

--- a/docs/generateAllConfigPartial.java
+++ b/docs/generateAllConfigPartial.java
@@ -1,5 +1,5 @@
 ///usr/bin/env jbang "$0" "$@" ; exit $?
-//DEPS org.jboss:jandex:2.4.3.Final
+//DEPS io.smallrye:jandex:3.1.8
 
 import org.jboss.jandex.AnnotationInstance;
 import org.jboss.jandex.DotName;

--- a/pom.xml
+++ b/pom.xml
@@ -850,7 +850,7 @@
                     <version>${version.checkstyle.plugin}</version>
                 </plugin>
                 <plugin>
-                    <groupId>org.jboss.jandex</groupId>
+                    <groupId>io.smallrye</groupId>
                     <artifactId>jandex-maven-plugin</artifactId>
                     <version>${jandex.version}</version>
                 </plugin>
@@ -924,7 +924,7 @@
         </pluginManagement>
         <plugins>
             <plugin>
-                <groupId>org.jboss.jandex</groupId>
+                <groupId>io.smallrye</groupId>
                 <artifactId>jandex-maven-plugin</artifactId>
                 <executions>
                     <execution>

--- a/pom.xml
+++ b/pom.xml
@@ -154,7 +154,7 @@
         <quarkus.version>3.8.4</quarkus.version>
 
         <!-- Jandex -->
-        <jandex.version>1.2.3</jandex.version>
+        <jandex.version>3.1.8</jandex.version>
 
         <!-- Schema types -->
         <avro.version>1.11.3</avro.version>

--- a/utils/exportConfluent/pom.xml
+++ b/utils/exportConfluent/pom.xml
@@ -81,7 +81,7 @@
             <plugin>
                 <groupId>io.smallrye</groupId>
                 <artifactId>jandex-maven-plugin</artifactId>
-                <version>1.2.3</version><!--$NO-MVN-MAN-VER$-->
+                <version>3.1.8</version><!--$NO-MVN-MAN-VER$-->
                 <executions>
                   <execution>
                     <id>make-index</id>

--- a/utils/exportConfluent/pom.xml
+++ b/utils/exportConfluent/pom.xml
@@ -79,7 +79,7 @@
             </plugin>
             
             <plugin>
-                <groupId>org.jboss.jandex</groupId>
+                <groupId>io.smallrye</groupId>
                 <artifactId>jandex-maven-plugin</artifactId>
                 <version>1.2.3</version><!--$NO-MVN-MAN-VER$-->
                 <executions>

--- a/utils/exportV1/pom.xml
+++ b/utils/exportV1/pom.xml
@@ -69,7 +69,7 @@
 
         <plugins>
             <plugin>
-                <groupId>org.jboss.jandex</groupId>
+                <groupId>io.smallrye</groupId>
                 <artifactId>jandex-maven-plugin</artifactId>
                 <version>1.2.3</version><!--$NO-MVN-MAN-VER$ -->
                 <executions>

--- a/utils/exportV1/pom.xml
+++ b/utils/exportV1/pom.xml
@@ -71,7 +71,7 @@
             <plugin>
                 <groupId>io.smallrye</groupId>
                 <artifactId>jandex-maven-plugin</artifactId>
-                <version>1.2.3</version><!--$NO-MVN-MAN-VER$ -->
+                <version>3.1.8</version><!--$NO-MVN-MAN-VER$ -->
                 <executions>
                     <execution>
                         <id>make-index</id>

